src/app/cube/user-profile/components/edit-profile/edit-profile.component.ts

```diff
@@ -0,0 +1,138 @@
+  ssoRedirect = environment.apiURL + '/google';
+  elementRef: any;
+  @Input() user;
+  @Input() gravatarImage;
+  @Output() close: EventEmitter<void> = new EventEmitter();
+  @Output() userInfo: EventEmitter<{}> = new EventEmitter();
+  errorMsg = 'There was an issue on our end with your registration, we are sorry for the inconvience.\n Please try again later!';
+  fieldErrorMsg = '';
+  editFailure: Boolean = false;
+  emailInUse = false;
+  loading = false;
+  editInfo = {
+    firstname: '',
+    lastname: '',
+    email: '',
+    organization: '',
+    bio: ''
+  };
+
+  editFormGroup: FormGroup = new FormGroup({
+    firstname: this.authValidation.getInputFormControl('required'),
+    lastname: this.authValidation.getInputFormControl('required'),
+    email: this.authValidation.getInputFormControl('email'),
+    organization: this.authValidation.getInputFormControl('required'),
+  });
+
+  constructor(
+    private authValidation: AuthValidationService,
+    private profileService: ProfileService,
+    private noteService: ToastrOvenService,
+    private auth: AuthService,
+    private router: Router
+  ) { }
+
+  ngOnChanges(): void {
+    this.authValidation.getErrorState().subscribe(err => this.editFailure = err);
+    this.editInfo = {
+      firstname: this.toUpper(this.user.name) ? this.toUpper(this.user.name.split(' ')[0]) : '',
+      lastname: this.toUpper(this.user.name) ? this.toUpper(this.user.name.substring(this.user.name.indexOf(' ') + 1)) : '',
+      email: this.user.email || '',
+      organization: this.toUpper(this.user.organization) || '',
+      bio: this.user.bio || ''
+    };
+  }
+
+  /**
+   * Saves User's information edits
+   *
+   * @private
+   * @memberof UserEditInformationComponent
+   */
+  async save() {
+    const edits = {
+      username: this.user.username,
+      name: `${this.editInfo.firstname.trim()} ${this.editInfo.lastname.trim()}`,
+      email: this.editInfo.email.trim(),
+      organization: this.editInfo.organization.trim(),
+      bio: this.editInfo.bio.trim(),
+    };
+    if(
+      this.editInfo.firstname !== ''
+      && this.editInfo.lastname !== ''
+      && this.editInfo.email !== ''
+      && this.editInfo.organization !== ''
+      && this.getValidEmail(this.editInfo.email)
+    ) {
+      try {
+        await this.profileService.editUserInfo(edits).then(async res => {
+          await this.auth.validateAndRefreshToken();
+          this.userInfo.next(edits);
+          this.close.next();
+          this.noteService.success('Success!', res);
+        });
+      } catch (e) {
+        // FIXME: No errors are being caught/thrown from user service
+        if (e.status === 400) {
+          this.noteService.error('Error!', e.error);
+        } else {
+          this.noteService.error('Error!', 'We couldn\'t update your user information!');
+        }
+      }
+    } else {
+      this.noteService.error('Error!', 'Double check the required fields above!');
+    }
```

Comment:
A suggestion to get rid of long if bodies
```suggestion
    if(
      this.editInfo.firstname === ''
      || this.editInfo.lastname === ''
      || this.editInfo.email === ''
      || this.editInfo.organization === ''
      || !this.getValidEmail(this.editInfo.email)
    ) {
      this.noteService.error('Error!', 'Double check the required fields above!');
      return;
    }
    try {
        await this.profileService.editUserInfo(edits).then(async res => {
          await this.auth.validateAndRefreshToken();
          this.userInfo.next(edits);
          this.close.next();
          this.noteService.success('Success!', res);
        });
      } catch (e) {
        // FIXME: No errors are being caught/thrown from user service
        if (e.status === 400) {
          this.noteService.error('Error!', e.error);
        } else {
          this.noteService.error('Error!', 'We couldn\'t update your user information!');
        }
      }
```

Reply:
You're not wrong, your way would work, the issue is the notification toaster doesn't throw or raise an error like we would in a service. The try catch would still execute but we don't want it to if any of the values are empty.

Reply:
Yeah i figured that. Thats why i added a `return` statement on line 88.

Reply:
I would rather do the and the the ors reason why is because the first false in the and will short it while the ors will check each expression in the if statement